File: server/src/main/java/org/dan/jadalnia/sys/async/AsynSync.java
package org.dan.jadalnia.sys.async;

import javax.ws.rs.container.AsyncResponse;
import java.util.concurrent.CompletableFuture;

public class AsynSync {
    public <T> void sync(
            CompletableFuture<T> resultFuture,
            AsyncResponse response) {
        resultFuture.whenComplete(
                (result, e) -> {
                    if (e == null) {
                        response.resume(result);
                    } else {
                        response.resume(e);
                    }
                });
    }
}

File: service/src/main/kotlin/org/dan/jadalnia/sys/async/AsynSync.kt
package org.dan.jadalnia.sys.async

import javax.ws.rs.container.AsyncResponse
import java.util.concurrent.CompletableFuture

class AsynSync {
    fun <T> sync(
            resultFuture: CompletableFuture<T>,
            response: AsyncResponse ) {
        resultFuture.whenComplete(
                { result, e ->
                    if (e == null) {
                        response.resume(result)
                    } else {
                        response.resume(e)
                    }
                })
    }
}
